src/NetscriptFunctions/Singularity.ts

```diff
@@ -1060,7 +1096,11 @@ export function NetscriptSingularity(
             workerScript.log("workForFaction", () => `Faction '${fac.name}' do not need help with hacking contracts.`);
             return false;
           }
-          player.startFactionHackWork(Router, fac);
+          player.startFactionHackWork(fac);
+          if (!wasWorking || (wasWorking && !wasFocused))
+            player.stopFocusing();
+          else if (wasWorking && wasFocused)
+            player.startFocusing();
```

Comment:
I think this is a backwards-compatiblity concern, users with "set work" scripts out there will expect it to focus the first time they set some work (which is required to gains full XP unless you have that aug). It looks like this code will *only* start focusing on the new work if they were previously working (and focused).

So I think you have to change this condition to just `if(wasFocused)` - which should be true by default (right?) when the game is loaded up and the player has never had a script explicitly call `setFocus(false)`?

Even then, we might get complaints that players relied on their scripts "putting them back into focused work" after some time if they popped by their computer, manually unfocused to check in on something, and left it unfocused. They would now need to edit their scripts to put them back in focus like they used to.

I'm not sure there's a backwards compatible way to do this unless we go with @hydroflame 's proposal to can the `setFocus` function entirely and instead add a new optional boolean `focus` parameter to each of the 3 work functions.

Reply:
Unfortunately, if the `workForFaction` or `workForCompany` command takes the focus by default, there is no storage of where the user was when focus was taken. If we then call setFocus(false) we have to send the user to some other page than the Work page (which becomes funky and invalid when isWorking false). If the Work screen was not a page but an overlay of the user's focus, this would work much better since we could just close the overlay, but that's beyond my limited React skills.

I originally proposed a focus parameter to the two work functions, so if that's the route that hydroflame wants to take I can refactor this PR.